tests/conftest.py

```diff
@@ -339,18 +339,28 @@ def generate_greedy_logprobs_limit(
                 processor_kwargs["images"] = images[i]
 
             inputs = self.processor(**processor_kwargs)
-            input_ids = inputs.input_ids
-
-            output = self.model.generate(
-                **self.wrap_device(inputs),
-                use_cache=True,
-                do_sample=False,
-                max_new_tokens=max_tokens,
-                output_hidden_states=True,
-                return_dict_in_generate=True,
-                **kwargs,
-            )
-
+            if "MiniCPM-Llama3-V-2_5" in self.model_name:
+                with torch.no_grad():
+                    output = self.model.generate(
+                        self.wrap_device(inputs),
+                        use_cache=True,
+                        do_sample=False,
+                        max_new_tokens=max_tokens,
+                        output_hidden_states=True,
+                        return_dict_in_generate=True,
+                        tokenizer=self.tokenizer,
+                        **kwargs,
+                    )
```

Comment:
Can we move this code to the model test file? You can pass the tokenizer via `kwargs`, and `torch.no_grad` can be entered outside of this function.

Reply:
There's another difference besides `torch.no_grad` which is the `inputs`...

Reply:
Could you elaborate?

Reply:
A slight difference
```python
            with torch.no_grad():
                if "MiniCPM-Llama3-V-2_5" in self.model_name:
                    output = self.model.generate(
                        self.wrap_device(inputs),
                        use_cache=True,
                        do_sample=False,
                        max_new_tokens=max_tokens,
                        output_hidden_states=True,
                        return_dict_in_generate=True,
                        tokenizer=self.tokenizer,
                        **kwargs,
                    )
                else:
                    output = self.model.generate(
                        **self.wrap_device(inputs),
                        use_cache=True,
                        do_sample=False,
                        max_new_tokens=max_tokens,
                        output_hidden_states=True,
                        return_dict_in_generate=True,
                        **kwargs,
                    )
```

Reply:
That is... awkward. Can you update the model on HF to support the same input style?

Reply:
Actually, let me commit to your branch. I have an idea that does not require modifying the HF model.